docker/Dockerfile

```diff
@@ -79,8 +79,8 @@ RUN pip install --no-cache-dir cmake ninja pybind11 wheel
 FROM base-build-python AS build-wheel-lightning-qubit
 WORKDIR /opt/pennylane-lightning
 RUN pip uninstall -y pennylane-lightning
-RUN python setup.py build_ext
-RUN python setup.py bdist_wheel
+RUN python scripts/configure_pyproject_toml.py
```

Comment:
Will [break](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10352814843) builds for previous versions. Not a big deal, but we have to be aware to build, say, v0.37.0 with the v0.37.0-version of the Dockerfile.

Reply:
So we probably need a boolean condition here. Any suggestions on how to better implement that?

Reply:
We could pass the current commit SHA and use `git merge-base`, but this is inconvenient I find. I'd say let's accept the breaking change for now.